extension/partiql-extension-ddl/tests/ddl-tests.rs

```diff
@@ -1,27 +1,27 @@
+use indexmap::IndexSet;
 use partiql_extension_ddl::ddl::{DdlFormat, PartiqlBasicDdlEncoder, PartiqlDdlEncoder};
 use partiql_types::{bag, int, r#struct, str, struct_fields, StructConstraint, StructField};
 use partiql_types::{BagType, PartiqlShape, Static, StructType};
-use std::collections::BTreeSet;
 
 #[test]
 fn basic_ddl_test() {
     let details_fields = struct_fields![("age", int!())];
-    let details = r#struct![BTreeSet::from([details_fields])];
+    let details = r#struct![IndexSet::from([details_fields])];
     let fields = [
         StructField::new("id", int!()),
         StructField::new("name", str!()),
         StructField::new("address", PartiqlShape::new_non_nullable(Static::String)),
         StructField::new_optional("details", details.clone()),
     ]
     .into();
-    let shape = bag![r#struct![BTreeSet::from([
+    let shape = bag![r#struct![IndexSet::from([
         StructConstraint::Fields(fields),
         StructConstraint::Open(false)
     ])]];
 
     let ddl_compact = PartiqlBasicDdlEncoder::new(DdlFormat::Compact);
     let actual = ddl_compact.ddl(&shape).expect("ddl_output");
-    let expected = r#""address" VARCHAR NOT NULL,"id" INT,"name" VARCHAR,"details" OPTIONAL STRUCT<"age": INT>"#;
+    let expected = r#""id" INT,"name" VARCHAR,"address" VARCHAR NOT NULL,"details" OPTIONAL STRUCT<"age": INT>"#;
```

Comment:
Note that it is the same set of fields and their types, but the order matches the order in which it was created.

E.g., 
```
    let fields = [
        StructField::new("id", int!()),
        StructField::new("name", str!()),
        StructField::new("address", PartiqlShape::new_non_nullable(Static::String)),
        StructField::new_optional("details", details.clone()),
    ]
```